python/lsst/ts/scheduler/model.py

```diff
@@ -1,44 +1,85 @@
 import time
+import logging
+
+from lsst.ts.scheduler import Driver
+from lsst.ts.scheduler.sal_utils import SALUtils
+
 from SALPY_scheduler import SAL_scheduler, scheduler_logevent_validSettingsC
+from scheduler_config.constants import CONFIG_DIRECTORY, CONFIG_DIRECTORY_PATH
+
+try:
+    from git import Repo
+except ImportError:
+    raise ImportError("gitpython not installed. Please install it with 'pip install gitpython' before proceeding.")
 
 DEFAULT_STATE = "OFFLINE"
 
 class Model():
-    """The model contains all business logic related to the scheduler.
-
-    The Model class is designed to solely interact with an interface object
-    called Driver. This is so that we can "plug and play" different scheduling
-    algorithms as long as they can interact with the Driver. Most of the methods
-    called on this class are called by the state machine, assuming we are in
-    operations mode. In order for any model to interact with the state machine
-    from ts_statemachine it must have an attribute "state". The "state" attribute is
-    used to store the current state of the state machine.
+    """A class that contains that contains the necessary attributes and methods
+    for a statemachine to control the Scheduling algorith.
+
+    Parameters
+    ----------
+    driver : ``module``
+        Optional driver for the Model to use. This allows external developers
+        to plug in thier own scheduling algorithms. External Driver's must
+        comply with the the Driver interface here 
+        https://docs.google.com/document/d/1wQ467QGqVP2e4iXOKcW2slLMgOzI_1Qk284b1gDGqaI
+
+    Notes
+    -----
+    The Model is the highest level interface for the State Machine to interact
+    with. The Scheduler State Machine has multiple states. Rather than having
+    to import many objects from different locations we import this Model object.
+    Allowing the Scheduler State Machine one unified object to make all its
+    calls from.
     """
 
-    def __init__(self):
+    def __init__(self, driver = Driver()):
```

Comment:
Can you add docstring to `__init__` method. It can be short or even a placeholder for documentation. Same applies for all other methods.

Reply:
I did have an __init__ docstring but I put it in the class level. I am assuming that perhaps we want to document our inits on the init method itself? Either works for me but we should stick to always doing it the same for our classes. 

I have using Google style for python documentation. 

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html